### PR TITLE
Use ccache 3.5.1

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: docker.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def get_numpy_choices():
 # ccache
 
 ccache = '''WORKDIR /opt/ccache
-RUN curl -L -s -o ccache.tar.gz https://github.com/ccache/ccache/archive/v3.4.2.tar.gz && \\
+RUN curl -L -s -o ccache.tar.gz https://github.com/ccache/ccache/archive/v3.6.0.tar.gz && \\
     tar -xzf ccache.tar.gz && cd ccache-3.4.2 && \\
```
Comment (gwtnb) [inline docker.py]: Is `cd ccache-3.4.2` OK?

Comment (okuta) [reply docker.py]: Thanks

**Comment by gwtnb** [comment]
Jenkins, test this please

**Comment by gwtnb** [comment]
It seems that 3.6.0 is invalid, so 3.6.0 should be replaced with 3.6. https://github.com/ccache/ccache/releases/tag/v3.6

**Comment by gwtnb** [comment]
Jenkins, test this please

**Comment by gwtnb** [comment]
Compiling ccache 3.6 fails. What is the purpose of upgrading ccache to 3.6? If updating to 3.6 is not mandatory, it seems good to use ccache 3.5.1 to save time.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by gwtnb** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
Test failures are not related to this PR.

**Pull request merged**